src/compile/data/calculate.ts

```diff
@@ -15,6 +18,32 @@ export class CalculateNode extends DataFlowNode {
     super();
   }
 
+  public static makeAllFromSort(model: ModelWithField): CalculateNode[] {
+    // get all the encoding with sort fields from model
+    const nodes = model.reduceFieldDef((acc: CalculateNode[], fieldDef: ChannelDef<any>) => {
+      if (isScaleFieldDef(fieldDef) && isSortArray(fieldDef.sort)) {
+        const transform: CalculateTransform = {
+          calculate: CalculateNode.calculateExpressionFromSortField(fieldDef.field, fieldDef.sort),
+          as: `${fieldDef.field}_sort_index`
```

Comment:
Should channel be a part of the name? 
(e.g., `${channel}_${fieldDef.field}_sort_index`.)

Imagine a troll case where I have x and y encode the same field, but use different `sort` array.  Your current code will behave incorrectly.